road/validation_road.py

```diff
@@ -0,0 +1,183 @@
+
+    # Part 3
+    freeflow_path = config['MAP_INPUT']['Freeflow_Dir']
+    shp_output_path = config['MAP_INPUT']['Shp_out_Dir']
+
+    data_file = config['MAP_YAML']['Csv_file']
+    shapes_file = config['MAP_YAML']['Shape_file']
+    join_field = config['MAP_YAML']['Join']
+    line_width_col = config['MAP_YAML']['Line_wid_col']
+    line_color_col = config['MAP_YAML']['Line_color_col']
+    breakpoints = config['MAP_YAML']['breakpoints']
+    dashboard_num3 = config['MAP_YAML']['Dashboard_number']
+    center = config['MAP_YAML']['Center']
+
+    # Part 1 - Scatter Plot
+    # Calculate the metrics
+    Select_time_period_loc_df, classification_col_types, file_name = compute_and_save_errors(
+        est_df, obs_df, chosen_timeperiod, combined_df_cols, classification_col)
+
+    # Generate the Scatter plots required files
+    row_count = 1
+    yaml_config = {'layout': {}}
+    for types in classification_col_types:
+        # Generate est vs obs scatter plot
+        generate_vega_lite_json_est(
+            file_name,
+            classification_col,
+            types,
+            x_field1,
+            y_field1,
+            fields1,
+            nominal_fields1,
+            name1)
+        # Generate percent errors vs obs scatter plot
+        generate_vega_lite_json_diffpercent(
+            file_name,
+            classification_col,
+            types,
+            x_field2,
+            y_field2,
+            fields2,
+            nominal_fields2,
+            name2)
+
+        yaml_config['layout'][f'row{row_count}'] = generate_yaml_config(
+            types, name1, name2)
+        row_count += 1
+
+    generate_vega_lite_json_est(
+        file_name,
+        classification_col,
+        'all',
+        x_field1,
+        y_field1,
+        fields1,
+        nominal_fields1,
+        name1,
+        include_all_data=True)
+
+    generate_vega_lite_json_diffpercent(
+        file_name,
+        classification_col,
+        'all',
+        x_field2,
+        y_field2,
+        fields2,
+        nominal_fields2,
+        name2,
+        include_all_data=True)
+
+    yaml_config['layout'][f'row{row_count}'] = generate_yaml_config(
+        'all', name1, name2, include_all_data=True)
+    yaml_file_path = f"dashboard{dashboard_num1}-Scatter.yaml"
+    with open(yaml_file_path, 'w') as file:
+        yaml.dump(yaml_config, file, default_flow_style=False)
+
+    # Part 2 - Validation Stats
+    time_period_dfs = prepare_time_period_dfs(
+        est_df, obs_df, times, combined_df_cols_stats)
+    generate_and_save_tables(time_period_dfs, group_vars)
+
+    # Part 3 - Map
+    merged_df = calculate_differences(est_df, obs_df)
+    process_geospatial_data(merged_df, freeflow_path, shp_output_path)
+    create_yaml_file(
+        center,
+        shapes_file,
+        data_file,
+        join_field,
+        line_width_col,
+        line_color_col,
+        breakpoints,
+        dashboard_num3)
+
+if __name__ == "__main__":
+    main()

```

Comment:
probably use argparse to read in the toml filename/filepath, and then pass in the toml filename/filepath into main()?